In [93]:
import pandas as pd
import numpy as np
import datetime as datetime
from datetime import date, timedelta
import matplotlib.pyplot as plt
import seaborn as sns
from math import radians, sin, cos, atan2, sqrt


%matplotlib inline
pd.options.display.max_rows = None
pd.options.display.max_columns = None

In [94]:
df_weather = pd.read_csv('../data/weather.csv')
df_spray = pd.read_csv('../data/spray.csv')
df_train = pd.read_csv('../data/train.csv')
df_test = pd.read_csv('../data/test.csv')

In [95]:
display(df_weather.head())
display(df_spray.head())
display(df_train.head())
display(df_test.head())

Station        Date  Tmax  Tmin Tavg Depart  DewPoint WetBulb Heat Cool  \
0        1  2007-05-01    83    50   67     14        51      56    0    2   
1        2  2007-05-01    84    52   68      M        51      57    0    3   
2        1  2007-05-02    59    42   51     -3        42      47   14    0   
3        2  2007-05-02    60    43   52      M        42      47   13    0   
4        1  2007-05-03    66    46   56      2        40      48    9    0   

  Sunrise Sunset CodeSum Depth Water1 SnowFall PrecipTotal StnPressure  \
0    0448   1849             0      M      0.0        0.00       29.10   
1       -      -             M      M        M        0.00       29.18   
2    0447   1850      BR     0      M      0.0        0.00       29.38   
3       -      -   BR HZ     M      M        M        0.00       29.44   
4    0446   1851             0      M      0.0        0.00       29.39   

  SeaLevel  ResultSpeed  ResultDir AvgSpeed  
0    29.82          1.7         27      9.2  
1    29.82          2.7         25      9.6  
2    30.09         13.0          4     13.4  
3    30.08         13.3          2     13.4  
4    30.12         11.7          7     11.9

Date        Time   Latitude  Longitude
0  2011-08-29  6:56:58 PM  42.391623 -88.089163
1  2011-08-29  6:57:08 PM  42.391348 -88.089163
2  2011-08-29  6:57:18 PM  42.391022 -88.089157
3  2011-08-29  6:57:28 PM  42.390637 -88.089158
4  2011-08-29  6:57:38 PM  42.390410 -88.088858

Date                                            Address  \
0  2007-05-29  4100 North Oak Park Avenue, Chicago, IL 60634,...   
1  2007-05-29  4100 North Oak Park Avenue, Chicago, IL 60634,...   
2  2007-05-29  6200 North Mandell Avenue, Chicago, IL 60646, USA   
3  2007-05-29    7900 West Foster Avenue, Chicago, IL 60656, USA   
4  2007-05-29    7900 West Foster Avenue, Chicago, IL 60656, USA   

                  Species  Block           Street  Trap  \
0  CULEX PIPIENS/RESTUANS     41   N OAK PARK AVE  T002   
1          CULEX RESTUANS     41   N OAK PARK AVE  T002   
2          CULEX RESTUANS     62    N MANDELL AVE  T007   
3  CULEX PIPIENS/RESTUANS     79     W FOSTER AVE  T015   
4          CULEX RESTUANS     79     W FOSTER AVE  T015   

              AddressNumberAndStreet   Latitude  Longitude  AddressAccuracy  \
0  4100  N OAK PARK AVE, Chicago, IL  41.954690 -87.800991                9   
1  4100  N OAK PARK AVE, Chicago, IL  41.954690 -87.800991                9   
2   6200  N MANDELL AVE, Chicago, IL  41.994991 -87.769279                9   
3    7900  W FOSTER AVE, Chicago, IL  41.974089 -87.824812                8   
4    7900  W FOSTER AVE, Chicago, IL  41.974089 -87.824812                8   

   NumMosquitos  WnvPresent  
0             1           0  
1             1           0  
2             1           0  
3             1           0  
4             4           0

Id        Date                                            Address  \
0   1  2008-06-11  4100 North Oak Park Avenue, Chicago, IL 60634,...   
1   2  2008-06-11  4100 North Oak Park Avenue, Chicago, IL 60634,...   
2   3  2008-06-11  4100 North Oak Park Avenue, Chicago, IL 60634,...   
3   4  2008-06-11  4100 North Oak Park Avenue, Chicago, IL 60634,...   
4   5  2008-06-11  4100 North Oak Park Avenue, Chicago, IL 60634,...   

                  Species  Block           Street  Trap  \
0  CULEX PIPIENS/RESTUANS     41   N OAK PARK AVE  T002   
1          CULEX RESTUANS     41   N OAK PARK AVE  T002   
2           CULEX PIPIENS     41   N OAK PARK AVE  T002   
3        CULEX SALINARIUS     41   N OAK PARK AVE  T002   
4         CULEX TERRITANS     41   N OAK PARK AVE  T002   

              AddressNumberAndStreet  Latitude  Longitude  AddressAccuracy  
0  4100  N OAK PARK AVE, Chicago, IL  41.95469 -87.800991                9  
1  4100  N OAK PARK AVE, Chicago, IL  41.95469 -87.800991                9  
2  4100  N OAK PARK AVE, Chicago, IL  41.95469 -87.800991                9  
3  4100  N OAK PARK AVE, Chicago, IL  41.95469 -87.800991                9  
4  4100  N OAK PARK AVE, Chicago, IL  41.95469 -87.800991                9

In [96]:
print('weather',df_weather.shape)
print('spray',df_spray.shape)
print('train',df_train.shape)
print('test',df_test.shape)

weather (2944, 22)
spray (14835, 4)
train (10506, 12)
test (116293, 11)


In [97]:
print(f'weather \n\n{df_weather.isnull().sum()}')
print(' ')
print(f'spray \n\n{df_spray.isnull().sum()}')
print(' ')
print(f'train \n\n{df_train.isnull().sum()}')
print(' ')
print(f'test \n\n{df_test.isnull().sum()}')

weather 

Station        0
Date           0
Tmax           0
Tmin           0
Tavg           0
Depart         0
DewPoint       0
WetBulb        0
Heat           0
Cool           0
Sunrise        0
Sunset         0
CodeSum        0
Depth          0
Water1         0
SnowFall       0
PrecipTotal    0
StnPressure    0
SeaLevel       0
ResultSpeed    0
ResultDir      0
AvgSpeed       0
dtype: int64
 
spray 

Date           0
Time         584
Latitude       0
Longitude      0
dtype: int64
 
train 

Date                      0
Address                   0
Species                   0
Block                     0
Street                    0
Trap                      0
AddressNumberAndStreet    0
Latitude                  0
Longitude                 0
AddressAccuracy           0
NumMosquitos              0
WnvPresent                0
dtype: int64
 
test 

Id                        0
Date                      0
Address                   0
Species                   0
Block                     0
Stre

In [98]:
df_spray.isnull().sum()

Date           0
Time         584
Latitude       0
Longitude      0
dtype: int64

In [99]:
print(f'weather \n\n{df_weather.dtypes}')
print(' ')
print(f'spray \n\n{df_spray.dtypes}')
print(' ')
print(f'train \n\n{df_train.dtypes}')
print(' ')
print(f'test \n\n{df_test.dtypes}')

weather 

Station          int64
Date            object
Tmax             int64
Tmin             int64
Tavg            object
Depart          object
DewPoint         int64
WetBulb         object
Heat            object
Cool            object
Sunrise         object
Sunset          object
CodeSum         object
Depth           object
Water1          object
SnowFall        object
PrecipTotal     object
StnPressure     object
SeaLevel        object
ResultSpeed    float64
ResultDir        int64
AvgSpeed        object
dtype: object
 
spray 

Date          object
Time          object
Latitude     float64
Longitude    float64
dtype: object
 
train 

Date                       object
Address                    object
Species                    object
Block                       int64
Street                     object
Trap                       object
AddressNumberAndStreet     object
Latitude                  float64
Longitude                 float64
AddressAccuracy             int64
NumMosquitos

In [100]:
# Change date type to datetime

In [101]:
df_weather['Date'] = pd.to_datetime(df_weather['Date'])
df_spray['Date'] = pd.to_datetime(df_spray['Date'])
df_train['Date'] = pd.to_datetime(df_train['Date'])
df_test['Date'] = pd.to_datetime(df_test['Date'])

In [102]:
print(f'weather \n\n{df_weather.dtypes}')
print(' ')
print(f'spray \n\n{df_spray.dtypes}')
print(' ')
print(f'train \n\n{df_train.dtypes}')
print(' ')
print(f'test \n\n{df_test.dtypes}')

weather 

Station                 int64
Date           datetime64[ns]
Tmax                    int64
Tmin                    int64
Tavg                   object
Depart                 object
DewPoint                int64
WetBulb                object
Heat                   object
Cool                   object
Sunrise                object
Sunset                 object
CodeSum                object
Depth                  object
Water1                 object
SnowFall               object
PrecipTotal            object
StnPressure            object
SeaLevel               object
ResultSpeed           float64
ResultDir               int64
AvgSpeed               object
dtype: object
 
spray 

Date         datetime64[ns]
Time                 object
Latitude            float64
Longitude           float64
dtype: object
 
train 

Date                      datetime64[ns]
Address                           object
Species                           object
Block                              int64
Street

In [103]:
# check timeline of all data

In [104]:
print('weather')
print('start:', df_weather['Date'].min())
print('end:',df_weather['Date'].max())
print(' ')
print('spray')
print('start:', df_spray['Date'].min())
print('end:',df_spray['Date'].max())
print(' ')
print('train')
print('start:', df_train['Date'].min())
print('end:',df_train['Date'].max())
print(' ')
print('test')
print('start:', df_test['Date'].min())
print('end:',df_test['Date'].max())

weather
start: 2007-05-01 00:00:00
end: 2014-10-31 00:00:00
 
spray
start: 2011-08-29 00:00:00
end: 2013-09-05 00:00:00
 
train
start: 2007-05-29 00:00:00
end: 2013-09-26 00:00:00
 
test
start: 2008-06-11 00:00:00
end: 2014-10-02 00:00:00


### Weather

Station 1: CHICAGO O'HARE INTERNATIONAL AIRPORT Lat: 41.995 Lon: -87.933 Elev: 662 ft. above sea level  
Station 2: CHICAGO MIDWAY INTL ARPT Lat: 41.786 Lon: -87.752 Elev: 612 ft. above sea level

In [105]:
df_weather.head()

Station       Date  Tmax  Tmin Tavg Depart  DewPoint WetBulb Heat Cool  \
0        1 2007-05-01    83    50   67     14        51      56    0    2   
1        2 2007-05-01    84    52   68      M        51      57    0    3   
2        1 2007-05-02    59    42   51     -3        42      47   14    0   
3        2 2007-05-02    60    43   52      M        42      47   13    0   
4        1 2007-05-03    66    46   56      2        40      48    9    0   

  Sunrise Sunset CodeSum Depth Water1 SnowFall PrecipTotal StnPressure  \
0    0448   1849             0      M      0.0        0.00       29.10   
1       -      -             M      M        M        0.00       29.18   
2    0447   1850      BR     0      M      0.0        0.00       29.38   
3       -      -   BR HZ     M      M        M        0.00       29.44   
4    0446   1851             0      M      0.0        0.00       29.39   

  SeaLevel  ResultSpeed  ResultDir AvgSpeed  
0    29.82          1.7         27      9.2  
1    29.82          2.7         25      9.6  
2    30.09         13.0          4     13.4  
3    30.08         13.3          2     13.4  
4    30.12         11.7          7     11.9

In [106]:
df_weather.shape

(2944, 22)

Fill in missing value with np.nan

In [107]:
df_weather = df_weather.replace('M', np.nan)

In [108]:
df_weather.head()

Station       Date  Tmax  Tmin Tavg Depart  DewPoint WetBulb Heat Cool  \
0        1 2007-05-01    83    50   67     14        51      56    0    2   
1        2 2007-05-01    84    52   68    NaN        51      57    0    3   
2        1 2007-05-02    59    42   51     -3        42      47   14    0   
3        2 2007-05-02    60    43   52    NaN        42      47   13    0   
4        1 2007-05-03    66    46   56      2        40      48    9    0   

  Sunrise Sunset CodeSum Depth  Water1 SnowFall PrecipTotal StnPressure  \
0    0448   1849             0     NaN      0.0        0.00       29.10   
1       -      -           NaN     NaN      NaN        0.00       29.18   
2    0447   1850      BR     0     NaN      0.0        0.00       29.38   
3       -      -   BR HZ   NaN     NaN      NaN        0.00       29.44   
4    0446   1851             0     NaN      0.0        0.00       29.39   

  SeaLevel  ResultSpeed  ResultDir AvgSpeed  
0    29.82          1.7         27      9.2  
1    29.82          2.7         25      9.6  
2    30.09         13.0          4     13.4  
3    30.08         13.3          2     13.4  
4    30.12         11.7          7     11.9

In [109]:
df_weather.dtypes

Station                 int64
Date           datetime64[ns]
Tmax                    int64
Tmin                    int64
Tavg                   object
Depart                 object
DewPoint                int64
WetBulb                object
Heat                   object
Cool                   object
Sunrise                object
Sunset                 object
CodeSum                object
Depth                  object
Water1                float64
SnowFall               object
PrecipTotal            object
StnPressure            object
SeaLevel               object
ResultSpeed           float64
ResultDir               int64
AvgSpeed               object
dtype: object

In [110]:
df_weather['PrecipTotal'].replace('  T', 0.005, inplace=True)

In [111]:
# convert 'object' type columns to numeric
df_weather['Tavg'] = pd.to_numeric(df_weather['Tavg'])
df_weather['Depart'] = pd.to_numeric(df_weather['Depart'])
df_weather['WetBulb'] = pd.to_numeric(df_weather['WetBulb'])
df_weather['Heat'] = pd.to_numeric(df_weather['Heat'])
df_weather['Cool'] = pd.to_numeric(df_weather['Cool'])
df_weather['PrecipTotal'] = pd.to_numeric(df_weather['PrecipTotal'])
df_weather['StnPressure'] = pd.to_numeric(df_weather['StnPressure'])
df_weather['SeaLevel'] = pd.to_numeric(df_weather['SeaLevel'])
df_weather['AvgSpeed'] = pd.to_numeric(df_weather['AvgSpeed'])

In [112]:
df_weather.dtypes

Station                 int64
Date           datetime64[ns]
Tmax                    int64
Tmin                    int64
Tavg                  float64
Depart                float64
DewPoint                int64
WetBulb               float64
Heat                  float64
Cool                  float64
Sunrise                object
Sunset                 object
CodeSum                object
Depth                  object
Water1                float64
SnowFall               object
PrecipTotal           float64
StnPressure           float64
SeaLevel              float64
ResultSpeed           float64
ResultDir               int64
AvgSpeed              float64
dtype: object

In [113]:
df_weather.isnull().sum()

Station           0
Date              0
Tmax              0
Tmin              0
Tavg             11
Depart         1472
DewPoint          0
WetBulb           4
Heat             11
Cool             11
Sunrise           0
Sunset            0
CodeSum           0
Depth          1472
Water1         2944
SnowFall       1472
PrecipTotal       2
StnPressure       4
SeaLevel          9
ResultSpeed       0
ResultDir         0
AvgSpeed          3
dtype: int64

#### Water1
all columns is missing

In [114]:
df_weather.drop(columns = 'Water1', axis=1, inplace=True)
df_weather.isnull().sum()

Station           0
Date              0
Tmax              0
Tmin              0
Tavg             11
Depart         1472
DewPoint          0
WetBulb           4
Heat             11
Cool             11
Sunrise           0
Sunset            0
CodeSum           0
Depth          1472
SnowFall       1472
PrecipTotal       2
StnPressure       4
SeaLevel          9
ResultSpeed       0
ResultDir         0
AvgSpeed          3
dtype: int64

#### Depart
Depart value is calculated by Tavg - 53 or 54

In [115]:
df_weather['Depart'][:10]

0    14.0
1     NaN
2    -3.0
3     NaN
4     2.0
5     NaN
6     4.0
7     NaN
8     5.0
9     NaN
Name: Depart, dtype: float64

In [116]:
df_weather['Depart'] = df_weather['Depart'].fillna((df_weather['Tavg'] - 53.5))

In [117]:
df_weather = df_weather.dropna(subset=['Depart'])

In [118]:
df_weather.isnull().sum()

Station           0
Date              0
Tmax              0
Tmin              0
Tavg              0
Depart            0
DewPoint          0
WetBulb           4
Heat              0
Cool              0
Sunrise           0
Sunset            0
CodeSum           0
Depth          1461
SnowFall       1461
PrecipTotal       2
StnPressure       4
SeaLevel          8
ResultSpeed       0
ResultDir         0
AvgSpeed          2
dtype: int64

#### Depth

In [119]:
df_weather['Depth'].value_counts()

0    1472
Name: Depth, dtype: int64

In [120]:
df_weather.drop(columns = 'Depth', axis=1, inplace=True)

#### Snowfall

In [121]:
df_weather['SnowFall'].value_counts()

0.0    1459
  T      12
0.1       1
Name: SnowFall, dtype: int64

In [122]:
df_weather.drop(columns = 'SnowFall', axis=1, inplace=True)

In [123]:
df_weather.isnull().sum()

Station        0
Date           0
Tmax           0
Tmin           0
Tavg           0
Depart         0
DewPoint       0
WetBulb        4
Heat           0
Cool           0
Sunrise        0
Sunset         0
CodeSum        0
PrecipTotal    2
StnPressure    4
SeaLevel       8
ResultSpeed    0
ResultDir      0
AvgSpeed       2
dtype: int64

#### WetBulb

In [129]:
df_weather['WetBulb'].isnull().sum()

0

In [128]:
df_weather['WetBulb'] = df_weather['WetBulb'].fillna(df_weather['WetBulb'].median())

#### PrecipTotal

In [130]:
df_weather['PrecipTotal'].isnull().sum()

2

In [132]:
df_weather[df_weather['PrecipTotal'].isnull()]

Station       Date  Tmax  Tmin  Tavg  Depart  DewPoint  WetBulb  Heat  \
117        2 2007-06-28    73    61  67.0    13.5        56     61.0   0.0   
119        2 2007-06-29    71    56  64.0    10.5        56     60.0   1.0   

     Cool Sunrise Sunset CodeSum  PrecipTotal  StnPressure  SeaLevel  \
117   2.0       -      -                  NaN        29.43     30.07   
119   0.0       -      -                  NaN        29.47     30.11   

     ResultSpeed  ResultDir  AvgSpeed  
117         12.2          2      13.3  
119          7.4          2       8.2

In [133]:
df_weather.drop(columns = 'PrecipTotal', axis=1, inplace=True)

#### StnPressure

In [134]:
df_weather['StnPressure'].isnull().sum()

4

In [135]:
df_weather[df_weather['StnPressure'].isnull()]

Station       Date  Tmax  Tmin  Tavg  Depart  DewPoint  WetBulb  Heat  \
87          2 2007-06-13    86    68  77.0    23.5        53     62.0   0.0   
848         1 2009-06-26    86    69  78.0     7.0        60     61.0   0.0   
2410        1 2013-08-10    81    64  73.0     0.0        57     61.0   0.0   
2411        2 2013-08-10    81    68  75.0    21.5        55     63.0   0.0   

      Cool Sunrise Sunset CodeSum  StnPressure  SeaLevel  ResultSpeed  \
87    12.0       -      -                  NaN       NaN          7.0   
848   13.0    0418   1931                  NaN     29.85          6.4   
2410   8.0    0454   1900                  NaN     30.08          5.3   
2411  10.0       -      -                  NaN     30.07          6.0   

      ResultDir  AvgSpeed  
87            5       NaN  
848           4       8.2  
2410          5       6.5  
2411          6       7.4

In [136]:
df_weather.drop(columns = 'StnPressure', axis=1, inplace=True)

#### SeaLevel

In [137]:
df_weather['SeaLevel'].isnull().sum()

8

In [138]:
df_weather[df_weather['SeaLevel'].isnull()]

Station       Date  Tmax  Tmin  Tavg  Depart  DewPoint  WetBulb  Heat  \
87          2 2007-06-13    86    68  77.0    23.5        53     62.0   0.0   
832         1 2009-06-18    80    61  71.0     1.0        63     67.0   0.0   
994         1 2009-09-07    77    59  68.0     1.0        59     62.0   0.0   
1732        1 2011-09-08    75    57  66.0     0.0        53     59.0   0.0   
1745        2 2011-09-14    60    48  54.0     0.5        45     51.0  11.0   
1756        1 2011-09-20    74    49  62.0     0.0        54     58.0   3.0   
2090        1 2012-09-03    88    71  80.0    12.0        70     73.0   0.0   
2743        2 2014-07-23    76    64  70.0    16.5        56     61.0   0.0   

      Cool Sunrise Sunset       CodeSum  SeaLevel  ResultSpeed  ResultDir  \
87    12.0       -      -                     NaN          7.0          5   
832    6.0    0416   1929         RA BR       NaN          6.7         16   
994    3.0    0523   1817            BR       NaN          5.8          3   
1732   1.0    0524   1815            RA       NaN         13.0          2   
1745   0.0       -      -   RA BR HZ FU       NaN          6.0         32   
1756   0.0    0537   1753  MIFG BCFG BR       NaN          7.3         18   
2090  15.0    0519   1824            BR       NaN          4.6          6   
2743   5.0       -      -                     NaN         16.4          2   

      AvgSpeed  
87         NaN  
832        7.9  
994        4.0  
1732      13.4  
1745       NaN  
1756       7.3  
2090       4.4  
2743      16.7

In [140]:
df_weather.drop(labels=['SeaLevel'],axis=1,inplace=True)

In [141]:
df_weather.isnull().sum()

Station        0
Date           0
Tmax           0
Tmin           0
Tavg           0
Depart         0
DewPoint       0
WetBulb        0
Heat           0
Cool           0
Sunrise        0
Sunset         0
CodeSum        0
ResultSpeed    0
ResultDir      0
AvgSpeed       2
dtype: int64

#### AvgSpeed

In [142]:
df_weather.drop(labels=['AvgSpeed'],axis=1,inplace=True)

In [143]:
df_weather.isnull().sum()

Station        0
Date           0
Tmax           0
Tmin           0
Tavg           0
Depart         0
DewPoint       0
WetBulb        0
Heat           0
Cool           0
Sunrise        0
Sunset         0
CodeSum        0
ResultSpeed    0
ResultDir      0
dtype: int64

In [144]:
df_weather.shape

(2933, 15)

In [164]:
df_weather.to_csv('../data/weather_cleaned.csv')

### Spray

In [145]:
df_spray.head()

Date        Time   Latitude  Longitude
0 2011-08-29  6:56:58 PM  42.391623 -88.089163
1 2011-08-29  6:57:08 PM  42.391348 -88.089163
2 2011-08-29  6:57:18 PM  42.391022 -88.089157
3 2011-08-29  6:57:28 PM  42.390637 -88.089158
4 2011-08-29  6:57:38 PM  42.390410 -88.088858

In [151]:
df_spray.dtypes

Date         datetime64[ns]
Time                 object
Latitude            float64
Longitude           float64
dtype: object

In [146]:
df_spray.shape

(14835, 4)

In [147]:
df_spray.isnull().sum()

Date           0
Time         584
Latitude       0
Longitude      0
dtype: int64

In [162]:
df_spray.drop(columns = 'Time', axis=1, inplace=True)
df_spray.isnull().sum()

Date         0
Latitude     0
Longitude    0
dtype: int64

In [148]:
df_spray.groupby('Date').count()

Time  Latitude  Longitude
Date                                 
2011-08-29    95        95         95
2011-09-07  1530      2114       2114
2013-07-17  2202      2202       2202
2013-07-25  1607      1607       1607
2013-08-08  1195      1195       1195
2013-08-15  2668      2668       2668
2013-08-16   141       141        141
2013-08-22  1587      1587       1587
2013-08-29  2302      2302       2302
2013-09-05   924       924        924

In [174]:
df_spray.duplicated().sum()

541

In [176]:
df_spray = df_spray.drop_duplicates()

In [177]:
df_spray.to_csv('../data/spray_cleaned.csv')

### Train

In [153]:
df_train.head()

Date                                            Address  \
0 2007-05-29  4100 North Oak Park Avenue, Chicago, IL 60634,...   
1 2007-05-29  4100 North Oak Park Avenue, Chicago, IL 60634,...   
2 2007-05-29  6200 North Mandell Avenue, Chicago, IL 60646, USA   
3 2007-05-29    7900 West Foster Avenue, Chicago, IL 60656, USA   
4 2007-05-29    7900 West Foster Avenue, Chicago, IL 60656, USA   

                  Species  Block           Street  Trap  \
0  CULEX PIPIENS/RESTUANS     41   N OAK PARK AVE  T002   
1          CULEX RESTUANS     41   N OAK PARK AVE  T002   
2          CULEX RESTUANS     62    N MANDELL AVE  T007   
3  CULEX PIPIENS/RESTUANS     79     W FOSTER AVE  T015   
4          CULEX RESTUANS     79     W FOSTER AVE  T015   

              AddressNumberAndStreet   Latitude  Longitude  AddressAccuracy  \
0  4100  N OAK PARK AVE, Chicago, IL  41.954690 -87.800991                9   
1  4100  N OAK PARK AVE, Chicago, IL  41.954690 -87.800991                9   
2   6200  N MANDELL AVE, Chicago, IL  41.994991 -87.769279                9   
3    7900  W FOSTER AVE, Chicago, IL  41.974089 -87.824812                8   
4    7900  W FOSTER AVE, Chicago, IL  41.974089 -87.824812                8   

   NumMosquitos  WnvPresent  
0             1           0  
1             1           0  
2             1           0  
3             1           0  
4             4           0

In [154]:
df_train.shape

(10506, 12)

In [179]:
df_train.drop(columns = 'Address', axis=1, inplace=True)
df_train.drop(columns = 'AddressNumberAndStreet', axis=1, inplace=True)

C:\Users\ACDMY2\anaconda3\lib\site-packages\pandas\core\frame.py:4163: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().drop(


In [180]:
df_train.isnull().sum()

Date               0
Species            0
Block              0
Street             0
Trap               0
Latitude           0
Longitude          0
AddressAccuracy    0
NumMosquitos       0
WnvPresent         0
dtype: int64

In [181]:
df_train.describe()

Block     Latitude    Longitude  AddressAccuracy  NumMosquitos  \
count  9693.000000  9693.000000  9693.000000      9693.000000   9693.000000   
mean     37.231714    41.847618   -87.702823         7.941092     10.210564   
std      24.330434     0.109416     0.093464         1.350981     13.138722   
min      10.000000    41.644612   -87.930995         3.000000      1.000000   
25%      13.000000    41.750498   -87.760886         8.000000      2.000000   
50%      36.000000    41.867108   -87.698457         8.000000      4.000000   
75%      58.000000    41.954690   -87.642984         9.000000     13.000000   
max      98.000000    42.017430   -87.531635         9.000000     50.000000   

        WnvPresent  
count  9693.000000  
mean      0.051893  
std       0.221823  
min       0.000000  
25%       0.000000  
50%       0.000000  
75%       0.000000  
max       1.000000

In [182]:
df_train.duplicated().sum()

0

In [183]:
df_train = df_train.drop_duplicates()

In [184]:
df_train.shape

(9693, 10)

In [185]:
df_train.to_csv('../data/train_cleaned.csv')

### Test

In [157]:
df_test.head()

Id       Date                                            Address  \
0   1 2008-06-11  4100 North Oak Park Avenue, Chicago, IL 60634,...   
1   2 2008-06-11  4100 North Oak Park Avenue, Chicago, IL 60634,...   
2   3 2008-06-11  4100 North Oak Park Avenue, Chicago, IL 60634,...   
3   4 2008-06-11  4100 North Oak Park Avenue, Chicago, IL 60634,...   
4   5 2008-06-11  4100 North Oak Park Avenue, Chicago, IL 60634,...   

                  Species  Block           Street  Trap  \
0  CULEX PIPIENS/RESTUANS     41   N OAK PARK AVE  T002   
1          CULEX RESTUANS     41   N OAK PARK AVE  T002   
2           CULEX PIPIENS     41   N OAK PARK AVE  T002   
3        CULEX SALINARIUS     41   N OAK PARK AVE  T002   
4         CULEX TERRITANS     41   N OAK PARK AVE  T002   

              AddressNumberAndStreet  Latitude  Longitude  AddressAccuracy  
0  4100  N OAK PARK AVE, Chicago, IL  41.95469 -87.800991                9  
1  4100  N OAK PARK AVE, Chicago, IL  41.95469 -87.800991                9  
2  4100  N OAK PARK AVE, Chicago, IL  41.95469 -87.800991                9  
3  4100  N OAK PARK AVE, Chicago, IL  41.95469 -87.800991                9  
4  4100  N OAK PARK AVE, Chicago, IL  41.95469 -87.800991                9

In [158]:
df_test.shape

(116293, 11)

In [186]:
df_test.drop(columns = 'Address', axis=1, inplace=True)
df_test.drop(columns = 'AddressNumberAndStreet', axis=1, inplace=True)

In [187]:
df_test.isnull().sum()

Id                 0
Date               0
Species            0
Block              0
Street             0
Trap               0
Latitude           0
Longitude          0
AddressAccuracy    0
dtype: int64

In [188]:
df_test.describe()

Id          Block       Latitude      Longitude  \
count  116293.000000  116293.000000  116293.000000  116293.000000   
mean    58147.000000      41.131100      41.849389     -87.693658   
std     33571.041765      24.864726       0.106593       0.080699   
min         1.000000      10.000000      41.644612     -87.930995   
25%     29074.000000      18.000000      41.753411     -87.750938   
50%     58147.000000      39.000000      41.862292     -87.694991   
75%     87220.000000      61.000000      41.951866     -87.648860   
max    116293.000000      98.000000      42.017430     -87.531635   

       AddressAccuracy  
count    116293.000000  
mean          7.954357  
std           1.252733  
min           3.000000  
25%           8.000000  
50%           8.000000  
75%           9.000000  
max           9.000000

In [189]:
df_test.to_csv('../data/test_cleaned.csv')